### PR TITLE
Fixes #31574: Ensure truststore certificates get updated when they change

### PR DESCRIPTION
**Pull request by ehelms**
This combines https://github.com/theforeman/puppet-certs/pull/312 with the updated testing from https://github.com/theforeman/puppet-certs/pull/311/files#diff-5afea89f9f4123c60bf87969c1a02b0d75c0687a14758480314204b44ca679c1R189 to get to a PR that resolves this issue and we can pull in. There have been additional reports of users hitting this issue.

### CODE REVIEW

**Comment by ehelms** [comment]
@ekohl @wbclark Could one of you look at and help me prioritize this? I've seen another two user reports this week where its being hit.

**Review by ekohl — APPROVED** [review]
Thanks!

**Comment by ekohl** [comment]
Not entirely sure how to classify it. It feels like a bugfix, but at the size that I wouldn't feel comfortable cherry picking it without careful consideration. Given the next is major anyway, it's not too critical but that's why I first chose enhancement and ended up with bug.

**Comment by ehelms** [comment]
:+1: to bug in that it definitely solves a problem that users run into -- thank you for getting back to this (and the original enhancement to my original PR that made it cleaner)

File: lib/puppet/type/truststore_certificate.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  newparam(:password_file) do
+    desc "Path to file containing the truststore password"
+    isrequired
+  end
+
+  autorequire(:file) do
+    [self[:password_file], File.dirname(self[:truststore]), self[:certificate]]
+  end
+
+  autonotify(:file) do
```
Comment (ekohl) [inline lib/puppet/type/truststore_certificate.rb:44]: Linking back: it was impossible to write a test for this using rspec-puppet. https://github.com/rodjek/rspec-puppet/commit/7484feb56ffa1380ffe778c7f9f89daa06c07570 has been merged but is unreleased. Once it's merged, it should be possible to test this:
```ruby
it { is_expected.to contain_truststore_certificate('the-cert').that_notifies('File[/truststore]') }
```

Comment (ehelms) [reply lib/puppet/type/truststore_certificate.rb:44]: Maybe file an issue to remember this? Or just open a draft PR with this added?

**Pull request merged**